src/components/Api.js

```diff
@@ -0,0 +1,114 @@
+export default class API {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error ${res.status}`);
+    }
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._handleResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
```

Comment:
**COULD BE IMPROVED**

Please, use your `_request` everywhere not to duplicate `.then(this._handleResponse)`

Reply:
Once I get the hang of refactoring the _request function, I plan to do so.